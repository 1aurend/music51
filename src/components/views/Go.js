import React from 'react'
import { Button } from 'shards-react'
import goSvg from '../../assets/svgs-go.svg'
import styled from 'styled-components'

const StyledSvgButton = styled(Button)`
  display: block;
  cursor: pointer;
  background-color: black;
  border: none;
  padding: 0;
`

export default function Go({ onClick }) {
  return (
    <>
      <StyledSvgButton
        theme="light"
        onClick={onClick}
        >
        <img src={goSvg} alt='go' style={{width: '8rem'}}/>
      </StyledSvgButton>
    </>
  )
}
