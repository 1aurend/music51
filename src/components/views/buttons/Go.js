import React from 'react'
import { Button } from 'shards-react'
import goSvg from '../../../assets/svgs-go.svg'
import styled from 'styled-components'


const StyledSvgButton = styled(Button)`
  display: block;
  cursor: pointer;
  border: none;
  padding: 0;
  background: transparent;
  :hover {
    background:transparent;
  }
`

export default function Go({ onClick }) {
  return (
    <>
      <StyledSvgButton
        theme="dark"
        onClick={onClick}
        >
        <img src={goSvg} alt='go' style={{width: '8rem'}}/>
      </StyledSvgButton>
    </>
  )
}
