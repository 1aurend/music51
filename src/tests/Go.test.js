import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Go from '../components/views/buttons/Go'


test('calls "generator" on click', () => {
  const generateQuiz = jest.fn(() => 'launching generator')
  const { asFragment, getByRole } = render(
  <Go onClick={generateQuiz}/>
  )
  fireEvent.click(getByRole('button'))
  expect(generateQuiz).toHaveBeenCalledTimes(1)
  expect(generateQuiz).toHaveReturnedWith('launching generator')
  // expect(asFragment()).toMatchSnapshot()
})
