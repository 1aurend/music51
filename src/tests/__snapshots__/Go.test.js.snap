// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calls "generator" on click 1`] = `
<DocumentFragment>
  .c0 {
  display: block;
  cursor: pointer;
  border: none;
  padding: 0;
}

<button
    class="c0 btn btn-dark"
    type="button"
  >
    <img
      alt="go"
      src="svgs-go.svg"
      style="width: 8rem;"
    />
  </button>
</DocumentFragment>
`;
